### PR TITLE
Use two-adic U256 fields for starks

### LINKED ISSUE #170: Switch prime field from U384 to U256 on Stark Proving System

### PR DESCRIPTION
**Pull request by mpaulucci**
# Use two-adic U256 fields for starks

## Description
Closes #170 
- We need fields with high adicity to use FFT. 
- This PR changes the primitive root of unity calculation to use the implementation from the `IsTwoAdicField`.

### CODE REVIEW

File: proving-system/stark/src/lib.rs [hunk 1]
```diff
@@ -11,51 +12,29 @@ use lambdaworks_math::polynomial::{self, Polynomial};
...
 use lambdaworks_math::{
-    field::fields::montgomery_backed_prime_fields::{IsModulus, U384PrimeField},
-    traits::ByteConversion,
-    unsigned_integer::element::U384,
+    field::fields::fft_friendly::u256_two_adic_prime_field::U256MontgomeryTwoAdicPrimeField,
+    traits::ByteConversion, unsigned_integer::element::U256,
 };
 
 // DEFINITION OF THE USED FIELD
 #[derive(Clone, Debug)]
```
Comment (IAvecilla) [inline proving-system/stark/src/lib.rs:20]: Can we delete this `derive`? I think it was meant for the old modulus struct

**Review by IAvecilla — APPROVED** [review]
LGTM! only some minor details

File: proving-system/stark/src/lib.rs [hunk 2]
```diff
@@ -332,12 +314,16 @@ pub fn verify_query(
...
+    let trace_primitive_root = PrimeField::get_primitive_root_of_unity(
+        ORDER_OF_ROOTS_OF_UNITY_TRACE.trailing_zeros() as u64,
+    )
+    .unwrap();
+    let lde_primitive_root = PrimeField::get_primitive_root_of_unity(
+        ORDER_OF_ROOTS_OF_UNITY_FOR_LDE.trailing_zeros() as u64,
+    )
+    .unwrap();
+
+    let zerofier = compute_zerofier(&trace_primitive_root, ORDER_OF_ROOTS_OF_UNITY_TRACE);
```
[the remark this answers is not in the document]
Comment (ilitteri) [reply proving-system/stark/src/lib.rs:326]: I agree, it'd be nice to track these and other `unwrap`s.

**Review by ilitteri — APPROVED** [review]

File: proving-system/stark/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpaulucci) [reply proving-system/stark/src/lib.rs:325]: I added one FIXME at the top of the prover function since there are so many `unwrap`s

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**